### PR TITLE
Refactor : Gateway Selector - Form Select to Select

### PR DESCRIPTION
**Pull request by besalraj30**
### Describe the change

Refactor: Use Select from @patternfly/react-core instead of FormSelect

### Steps to test the PR 

Render GatewaySelector component

Issue : https://github.com/kiali/kiali/issues/7277

image

### CODE REVIEW

File: frontend/src/components/IstioWizards/GatewaySelector.tsx [hunk 1]
```diff
@@ -57,10 +60,33 @@ export class GatewaySelector extends React.Component<Props, GatewaySelectorState
...
+    this.setState({ selectedGateway: value as string, isOpen: false });
+  };
+
+  toggle = (toggleRef: React.Ref<any>) => (
+    <MenuToggle
+      ref={toggleRef}
+      onClick={this.onToggleClick}
+      isExpanded={this.state.isOpen} 
+      isDisabled={!this.state.addGateway || this.state.newGateway || this.props.gateways.length === 0} 
+      style={{
```
Comment (ferhoyos) [inline frontend/src/components/IstioWizards/GatewaySelector.tsx:82]: I think it is preferable to remove the fix width, so the select component adjusts to the width of the content

File: frontend/src/components/IstioWizards/GatewaySelector.tsx [hunk 2]
```diff
@@ -217,20 +243,29 @@ export class GatewaySelector extends React.Component<Props, GatewaySelectorState
...
-                    value={this.state.selectedGateway}
-                    isDisabled={!this.state.addGateway || this.state.newGateway || this.props.gateways.length === 0}
-                    onChange={(_event, gw: string) => this.onFormChange(GatewayForm.GATEWAY_SELECTED, gw)}
-                  >
+              {this.props.gateways.length > 0 && (
+                <Select
+                  id="selectGateway"
+                  isOpen={this.state.isOpen}
+                  selected={this.state.selectedGateway}
+                  onSelect={this.onSelect}
```
Comment (ferhoyos) [inline frontend/src/components/IstioWizards/GatewaySelector.tsx:251]: you need to call `this.onFormChange` function to maintain the same application behaviour. In this case, you will have to add `isOpen: false` to the `this.onFormChange` function:

```
case GatewayForm.GATEWAY_SELECTED:
        this.setState(
          {
            selectedGateway: value,
            isOpen: false
          },
          () => this.props.onGatewayChange(this.isGatewayValid(), this.state)
        );
        break;
```

```suggestion
                  onSelect={(_event, gw) => this.onFormChange(GatewayForm.GATEWAY_SELECTED, gw as string)}
```

File: frontend/src/components/IstioWizards/GatewaySelector.tsx [hunk 3]
```diff
@@ -57,10 +60,33 @@ export class GatewaySelector extends React.Component<Props, GatewaySelectorState
...
+      port: 80,
+      isOpen: false,
     };
   }
 
+  onToggleClick = () => {
+    this.setState((prevState) => ({ isOpen: !prevState.isOpen }));
+  };
+
+  onSelect = (_event: React.MouseEvent<Element, MouseEvent> | undefined, value: string | number | undefined) => {
```
Comment (ferhoyos) [inline frontend/src/components/IstioWizards/GatewaySelector.tsx:72]: This function is no longer needed

**Review by ferhoyos — CHANGES_REQUESTED** [review]
Thank you very much for your PR @besalraj30! Overall, the refactoring is excellent. Just a few comments need to be addressed before approval.

**Comment by hhovsepy** [comment]
For Gateway selector (non K8sGateway), the chosen Gateway is not selected in fact:

When you have several gateways, and editing the existing traffic via wizzard, chose another gateway:
Screenshot from 2024-10-14 10-35-24
Still the old one is kept:
Screenshot from 2024-10-14 10-35-44

File: frontend/src/components/IstioWizards/GatewaySelector.tsx (hunk 1 above)
Comment (besalraj30) [reply frontend/src/components/IstioWizards/GatewaySelector.tsx]: resolved

File: frontend/src/components/IstioWizards/GatewaySelector.tsx (hunk 2 above)
Comment (besalraj30) [reply frontend/src/components/IstioWizards/GatewaySelector.tsx]: resolved

File: frontend/src/components/IstioWizards/GatewaySelector.tsx (hunk 3 above)
Comment (besalraj30) [reply frontend/src/components/IstioWizards/GatewaySelector.tsx]: resolved

**Comment by besalraj30** [comment]
> For Gateway selector (non K8sGateway), the chosen Gateway is not selected in fact:
> 
> When you have several gateways, and editing the existing traffic via wizzard, chose another gateway: Screenshot from 2024-10-14 10-35-24 Still the old one is kept: Screenshot from 2024-10-14 10-35-44



> For Gateway selector (non K8sGateway), the chosen Gateway is not selected in fact:
> 
> When you have several gateways, and editing the existing traffic via wizzard, chose another gateway: Screenshot from 2024-10-14 10-35-24 Still the old one is kept: Screenshot from 2024-10-14 10-35-44

Resolved this issue

**Review by ferhoyos — APPROVED** [review]
LGTM. Well done, thanks for your contribution @besalraj30! Looking forward to reviewing more `Select` refactor PRs.

**Review by hhovsepy — APPROVED** [review]
verified, thank you @besalraj30

**Pull request merged**